org.eclipse.lsp4e.jdt/src/org/eclipse/lsp4e/jdt/LSJavaProposal.java

```diff
@@ -94,18 +84,97 @@ public int getRelevance() {
 					break;
 				default:
 				}
-			}
 			relevanceComputed = true;
 		}
 		return relevance;
 	}
 	
-	private int computeBaseRelevance(LSCompletionProposal c) {
+	private int computeBaseRelevance() {
 		// Incorporate LSP4E category and rank into base relevance.
-		int base = MAX_BASE_RELEVANCE - (c.getRankCategory() - 1) * RANGE_WITHIN_CATEGORY;
-		int rank = c.getRankScore();
+		int base = MAX_BASE_RELEVANCE - (getRankCategory() - 1) * RANGE_WITHIN_CATEGORY;
+		int rank = getRankScore();
 		base -= (rank >= 0 && rank < RANGE_WITHIN_CATEGORY ? rank : RANGE_WITHIN_CATEGORY);
 		return base;
 	}
+	
+	/*
+	 * Copied from org.eclipse.jdt.internal.ui.text.java.AbstractJavaCompletionProposal.getInformationControlCreator()
+	 */
+	@Override
+	public @Nullable IInformationControlCreator getInformationControlCreator() {
+		Shell shell= JavaPlugin.getActiveWorkbenchShell();
+		if (shell == null || !BrowserInformationControl.isAvailable(shell))
+			return null;
+
+		if (infoControlCreator == null) {
+			/*
+			 * FIXME: Take control creators (and link handling) out of JavadocHover,
```

Comment:
This has been marked as RESOLVED WONTFIX, so probably it will never happen. Should we remove the FIXME and leave the link to the bug report?

Reply:
Remove part of the comment